### PR TITLE
Misc fixes with validation, version, clustering

### LINKED ISSUE #57: Ensure number of contact points set to replica count

### PR DESCRIPTION
**Pull request by longshorej**
* Modifies the CLI to use scalaz's `ValidationNel` so that failures can be accumulated and shown at once.
* Adds a validation for Akka Cluster Bootstrapping to ensure that a proper `--pod-controller-replicas` value has been specified.
* Ensures that `-Dakka.cluster.bootstrap.contact-point-discovery.required-contact-point-nr` is set to the number of replicas. 
* Ensures that `-Dakka.cluster.bootstrap.contact-point-discovery.effective-name` is set
* Simplify versioning to align with https://github.com/lightbend/sbt-reactive-app/pull/42. We really only need two naming types - `appName` (rolling) and `appNameVersion` (canary, blue/green). Thus, we can support any arbitrary version string, there is no need to require the user to use semantic versioning. Of course, we should recommend they use it in the documentation. Adding to this, the default version for an SBT project is not semantic, so prior to this change our tooling would fail for most people.

Fixes #57

### CODE REVIEW

**Comment by longshorej** [comment]
Examples

```bash
$ reactive-cli-out generate-kubernetes-deployment mysql:8.0.3 
[error, com.lightbend.rp.reactivecli.Main$] Docker label "com.lightbend.rp.app-name" must be defined
[error, com.lightbend.rp.reactivecli.Main$] Docker label "com.lightbend.rp.version" must be defined
-> 0

$ reactive-cli-out generate-kubernetes-deployment activator-lagom-java-chirper/friend-impl:1.0.0-SNAPSHOT  --pod-controller-replicas 1
[error, com.lightbend.rp.reactivecli.Main$] Akka Cluster Bootstrapping is enabled so you must specify `--pod-controller-replicas 2` (or greater)
-> 0

```

File: cli/src/main/scala/com/lightbend/rp/reactivecli/docker/DockerRegistry.scala [hunk 1]
```diff
@@ -106,12 +105,12 @@ object DockerRegistry extends LazyLogging {
   private def getWithToken(http: HttpExchange, credentials: Option[HttpRequest.BasicAuth], validateTls: Boolean)(url: String, headers: HttpHeaders, tryNewToken: Boolean = true, token: Option[HttpRequest.BearerToken] = None): Try[(HttpResponse, Option[HttpRequest.BearerToken])] = {
     val request =
       HttpRequest(url)
-        .headers(token.fold(headers)(t => headers.updated("Authorization", s"Bearer $t")))
+        .headers(token.fold(headers)(t => headers.updated("Authorization", s"Bearer ${t.value}")))
```
Comment (fsat) [inline cli/src/main/scala/com/lightbend/rp/reactivecli/docker/DockerRegistry.scala]: Oops... thanks for fixing this! 👍

File: cli/src/main/scala/com/lightbend/rp/reactivecli/runtime/kubernetes/Deployment.scala [hunk 2]
```diff
@@ -127,6 +139,24 @@ object Deployment {
         }
         .toMap
 
+    private[kubernetes] def mergeEnvs(envs: Map[String, EnvironmentVariable]*): Map[String, EnvironmentVariable] = {
+      val concatLiteral = Set("RP_JAVA_OPTS")
```
Comment (fsat) [inline cli/src/main/scala/com/lightbend/rp/reactivecli/runtime/kubernetes/Deployment.scala]: Just a minor thing, but I'd put `concatLiteral` as a constant in the `Deployment. RpEnvironmentVariables` object called `concatLiteralEnvironmentVariables`

File: cli/src/main/scala/com/lightbend/rp/reactivecli/runtime/kubernetes/Service.scala [hunk 3]
```diff
@@ -54,44 +57,39 @@ object Service {
...
-                    "ports" -> annotations.endpoints.asJson,
-                    "selector" -> Json(selector))))))
-        case _ =>
-          Failure(new IllegalArgumentException("Unable to generate Service resource: application name is required"))
-      }
+      val selector =
+        deploymentType match {
+          case CanaryDeploymentType    => "app" -> appName.asJson
+          case RollingDeploymentType   => "app" -> appName.asJson
+          case BlueGreenDeploymentType => "appVersion" -> appVersion.asJson
```
Comment (fsat) [inline cli/src/main/scala/com/lightbend/rp/reactivecli/runtime/kubernetes/Service.scala]: Can I confirm for `BlueGreenDeploymentType` we are only checking the app version? This contains only the semver, i.e. `3.2.1` - so I do think we need the `appName` as well:

```
val selector =
        deploymentType match {
           case CanaryDeploymentType    => Json("app" -> appName.asJson)
           case RollingDeploymentType   => Json("app" -> appName.asJson)
           case BlueGreenDeploymentType => Json("app" -> appName.asJson, "appVersion" -> appVersion.asJson)
         }
```

File: cli/src/main/scala/com/lightbend/rp/reactivecli/runtime/kubernetes/Service.scala [hunk 4]
```diff
@@ -54,44 +57,39 @@ object Service {
...
+              "metadata" -> Json(
+                "labels" -> Json(
+                  "app" -> appName.asJson),
+                "name" -> appName.asJson)
+                .deepmerge(
+                  annotations.namespace.fold(jEmptyObject)(ns => Json("namespace" -> serviceName(ns).asJson))),
+              "spec" -> Json(
+                "clusterIP" -> clusterIp.getOrElse("None").asJson,
+                "ports" -> annotations.endpoints.asJson,
+                "selector" -> Json(selector)))))
```
Comment (fsat) [inline cli/src/main/scala/com/lightbend/rp/reactivecli/runtime/kubernetes/Service.scala]: If we do need the `appName` in the `BlueGreenDeploymentType`, this needs to be modified:

```
 "selector" -> selector
```

File: cli/src/main/scala/com/lightbend/rp/reactivecli/runtime/kubernetes/Service.scala (hunk 3 above)
Comment (longshorej) [reply cli/src/main/scala/com/lightbend/rp/reactivecli/runtime/kubernetes/Service.scala]: `appVersion` has `appName` in it .. it should probably be named `appNameVersion`.

**Comment by longshorej** [comment]
Here's what the resources looking like for the three deployment types:

# Canary

```json
{
  "apiVersion" : "apps/v1beta2",
  "kind" : "Deployment",
  "metadata" : {
    "name" : "akka-cluster-tooling-example-v0-1-0",
    "labels" : {
      "appName" : "akka-cluster-tooling-example",
      "appNameVersion" : "akka-cluster-tooling-example-v0-1-0"
    },
    "namespace" : "akka-cluster-tooling-example"
  },
  "spec" : {
    "replicas" : 3,
    "selector" : {
      "matchLabels" : {
        "appNameVersion" : "akka-cluster-tooling-example-v0-1-0"
      }
    },
    "template" : {
      "metadata" : {
        "labels" : {
          "appName" : "akka-cluster-tooling-example",
          "appNameVersion" : "akka-cluster-tooling-example-v0-1-0"
        }
      },
      "spec" : {
        "containers" : [
          {
            "name" : "akka-cluster-tooling-example",
            "image" : "akka-cluster-tooling-example/akka-cluster-tooling-example:0.1.0",
            "imagePullPolicy" : "IfNotPresent",
            "env" : [
              {
                "name" : "RP_APP_NAME",
                "value" : "akka-cluster-tooling-example"
              },
              {
                "name" : "RP_APP_TYPE",
                "value" : "basic"
              },
              {
                "name" : "RP_APP_VERSION",
                "value" : "0.1.0"
              },
              {
                "name" : "RP_ENDPOINTS",
                "value" : "AKKA_REMOTE,AKKA_MGMT_HTTP,HTTP"
              },
              {
                "name" : "RP_ENDPOINTS_COUNT",
                "value" : "3"
              },
              {
                "name" : "RP_ENDPOINT_0_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_0_BIND_PORT",
                "value" : "10000"
              },
              {
                "name" : "RP_ENDPOINT_0_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_0_PORT",
                "value" : "10000"
              },
              {
                "name" : "RP_ENDPOINT_1_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_1_BIND_PORT",
                "value" : "10001"
              },
              {
                "name" : "RP_ENDPOINT_1_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_1_PORT",
                "value" : "10001"
              },
              {
                "name" : "RP_ENDPOINT_2_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_2_BIND_PORT",
                "value" : "10002"
              },
              {
                "name" : "RP_ENDPOINT_2_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_2_PORT",
                "value" : "10002"
              },
              {
                "name" : "RP_ENDPOINT_AKKA_MGMT_HTTP_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_AKKA_MGMT_HTTP_BIND_PORT",
                "value" : "10001"
              },
              {
                "name" : "RP_ENDPOINT_AKKA_MGMT_HTTP_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_AKKA_MGMT_HTTP_PORT",
                "value" : "10001"
              },
              {
                "name" : "RP_ENDPOINT_AKKA_REMOTE_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_AKKA_REMOTE_BIND_PORT",
                "value" : "10000"
              },
              {
                "name" : "RP_ENDPOINT_AKKA_REMOTE_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_AKKA_REMOTE_PORT",
                "value" : "10000"
              },
              {
                "name" : "RP_ENDPOINT_HTTP_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_HTTP_BIND_PORT",
                "value" : "10002"
              },
              {
                "name" : "RP_ENDPOINT_HTTP_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_HTTP_PORT",
                "value" : "10002"
              },
              {
                "name" : "RP_JAVA_OPTS",
                "value" : "-Dakka.cluster.bootstrap.contact-point-discovery.effective-name=akka-cluster-tooling-example -Dakka.cluster.bootstrap.contact-point-discovery.required-contact-point-nr=3"
              },
              {
                "name" : "RP_KUBERNETES_POD_IP",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_KUBERNETES_POD_NAME",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "metadata.name"
                  }
                }
              },
              {
                "name" : "RP_MODULES",
                "value" : "akka-cluster-bootstrapping,common,service-discovery"
              },
              {
                "name" : "RP_NAMESPACE",
                "value" : "akka-cluster-tooling-example"
              },
              {
                "name" : "RP_PLATFORM",
                "value" : "kubernetes"
              }
            ],
            "ports" : [
              {
                "containerPort" : 10000,
                "name" : "akka-remote"
              },
              {
                "containerPort" : 10001,
                "name" : "akka-mgmt-http"
              },
              {
                "containerPort" : 10002,
                "name" : "http"
              }
            ]
          }
        ]
      }
    }
  }
}

```

# Rolling

```json
{
  "apiVersion" : "apps/v1beta2",
  "kind" : "Deployment",
  "metadata" : {
    "name" : "akka-cluster-tooling-example",
    "labels" : {
      "appName" : "akka-cluster-tooling-example",
      "appNameVersion" : "akka-cluster-tooling-example-v0-1-0"
    },
    "namespace" : "akka-cluster-tooling-example"
  },
  "spec" : {
    "replicas" : 3,
    "selector" : {
      "matchLabels" : {
        "appName" : "akka-cluster-tooling-example"
      }
    },
    "template" : {
      "metadata" : {
        "labels" : {
          "appName" : "akka-cluster-tooling-example",
          "appNameVersion" : "akka-cluster-tooling-example-v0-1-0"
        }
      },
      "spec" : {
        "containers" : [
          {
            "name" : "akka-cluster-tooling-example",
            "image" : "akka-cluster-tooling-example/akka-cluster-tooling-example:0.1.0",
            "imagePullPolicy" : "IfNotPresent",
            "env" : [
              {
                "name" : "RP_APP_NAME",
                "value" : "akka-cluster-tooling-example"
              },
              {
                "name" : "RP_APP_TYPE",
                "value" : "basic"
              },
              {
                "name" : "RP_APP_VERSION",
                "value" : "0.1.0"
              },
              {
                "name" : "RP_ENDPOINTS",
                "value" : "AKKA_REMOTE,AKKA_MGMT_HTTP,HTTP"
              },
              {
                "name" : "RP_ENDPOINTS_COUNT",
                "value" : "3"
              },
              {
                "name" : "RP_ENDPOINT_0_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_0_BIND_PORT",
                "value" : "10000"
              },
              {
                "name" : "RP_ENDPOINT_0_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_0_PORT",
                "value" : "10000"
              },
              {
                "name" : "RP_ENDPOINT_1_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_1_BIND_PORT",
                "value" : "10001"
              },
              {
                "name" : "RP_ENDPOINT_1_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_1_PORT",
                "value" : "10001"
              },
              {
                "name" : "RP_ENDPOINT_2_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_2_BIND_PORT",
                "value" : "10002"
              },
              {
                "name" : "RP_ENDPOINT_2_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_2_PORT",
                "value" : "10002"
              },
              {
                "name" : "RP_ENDPOINT_AKKA_MGMT_HTTP_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_AKKA_MGMT_HTTP_BIND_PORT",
                "value" : "10001"
              },
              {
                "name" : "RP_ENDPOINT_AKKA_MGMT_HTTP_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_AKKA_MGMT_HTTP_PORT",
                "value" : "10001"
              },
              {
                "name" : "RP_ENDPOINT_AKKA_REMOTE_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_AKKA_REMOTE_BIND_PORT",
                "value" : "10000"
              },
              {
                "name" : "RP_ENDPOINT_AKKA_REMOTE_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_AKKA_REMOTE_PORT",
                "value" : "10000"
              },
              {
                "name" : "RP_ENDPOINT_HTTP_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_HTTP_BIND_PORT",
                "value" : "10002"
              },
              {
                "name" : "RP_ENDPOINT_HTTP_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_HTTP_PORT",
                "value" : "10002"
              },
              {
                "name" : "RP_JAVA_OPTS",
                "value" : "-Dakka.cluster.bootstrap.contact-point-discovery.effective-name=akka-cluster-tooling-example -Dakka.cluster.bootstrap.contact-point-discovery.required-contact-point-nr=3"
              },
              {
                "name" : "RP_KUBERNETES_POD_IP",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_KUBERNETES_POD_NAME",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "metadata.name"
                  }
                }
              },
              {
                "name" : "RP_MODULES",
                "value" : "akka-cluster-bootstrapping,common,service-discovery"
              },
              {
                "name" : "RP_NAMESPACE",
                "value" : "akka-cluster-tooling-example"
              },
              {
                "name" : "RP_PLATFORM",
                "value" : "kubernetes"
              }
            ],
            "ports" : [
              {
                "containerPort" : 10000,
                "name" : "akka-remote"
              },
              {
                "containerPort" : 10001,
                "name" : "akka-mgmt-http"
              },
              {
                "containerPort" : 10002,
                "name" : "http"
              }
            ]
          }
        ]
      }
    }
  }
}

```

```json
{
  "apiVersion" : "apps/v1beta2",
  "kind" : "Deployment",
  "metadata" : {
    "name" : "akka-cluster-tooling-example-v0-1-0",
    "labels" : {
      "appName" : "akka-cluster-tooling-example",
      "appNameVersion" : "akka-cluster-tooling-example-v0-1-0"
    },
    "namespace" : "akka-cluster-tooling-example"
  },
  "spec" : {
    "replicas" : 3,
    "selector" : {
      "matchLabels" : {
        "appNameVersion" : "akka-cluster-tooling-example-v0-1-0"
      }
    },
    "template" : {
      "metadata" : {
        "labels" : {
          "appName" : "akka-cluster-tooling-example",
          "appNameVersion" : "akka-cluster-tooling-example-v0-1-0"
        }
      },
      "spec" : {
        "containers" : [
          {
            "name" : "akka-cluster-tooling-example",
            "image" : "akka-cluster-tooling-example/akka-cluster-tooling-example:0.1.0",
            "imagePullPolicy" : "IfNotPresent",
            "env" : [
              {
                "name" : "RP_APP_NAME",
                "value" : "akka-cluster-tooling-example"
              },
              {
                "name" : "RP_APP_TYPE",
                "value" : "basic"
              },
              {
                "name" : "RP_APP_VERSION",
                "value" : "0.1.0"
              },
              {
                "name" : "RP_ENDPOINTS",
                "value" : "AKKA_REMOTE,AKKA_MGMT_HTTP,HTTP"
              },
              {
                "name" : "RP_ENDPOINTS_COUNT",
                "value" : "3"
              },
              {
                "name" : "RP_ENDPOINT_0_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_0_BIND_PORT",
                "value" : "10000"
              },
              {
                "name" : "RP_ENDPOINT_0_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_0_PORT",
                "value" : "10000"
              },
              {
                "name" : "RP_ENDPOINT_1_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_1_BIND_PORT",
                "value" : "10001"
              },
              {
                "name" : "RP_ENDPOINT_1_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_1_PORT",
                "value" : "10001"
              },
              {
                "name" : "RP_ENDPOINT_2_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_2_BIND_PORT",
                "value" : "10002"
              },
              {
                "name" : "RP_ENDPOINT_2_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_2_PORT",
                "value" : "10002"
              },
              {
                "name" : "RP_ENDPOINT_AKKA_MGMT_HTTP_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_AKKA_MGMT_HTTP_BIND_PORT",
                "value" : "10001"
              },
              {
                "name" : "RP_ENDPOINT_AKKA_MGMT_HTTP_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_AKKA_MGMT_HTTP_PORT",
                "value" : "10001"
              },
              {
                "name" : "RP_ENDPOINT_AKKA_REMOTE_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_AKKA_REMOTE_BIND_PORT",
                "value" : "10000"
              },
              {
                "name" : "RP_ENDPOINT_AKKA_REMOTE_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_AKKA_REMOTE_PORT",
                "value" : "10000"
              },
              {
                "name" : "RP_ENDPOINT_HTTP_BIND_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_HTTP_BIND_PORT",
                "value" : "10002"
              },
              {
                "name" : "RP_ENDPOINT_HTTP_HOST",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_ENDPOINT_HTTP_PORT",
                "value" : "10002"
              },
              {
                "name" : "RP_JAVA_OPTS",
                "value" : "-Dakka.cluster.bootstrap.contact-point-discovery.effective-name=akka-cluster-tooling-example-v0-1-0 -Dakka.cluster.bootstrap.contact-point-discovery.required-contact-point-nr=3"
              },
              {
                "name" : "RP_KUBERNETES_POD_IP",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "status.podIP"
                  }
                }
              },
              {
                "name" : "RP_KUBERNETES_POD_NAME",
                "valueFrom" : {
                  "fieldRef" : {
                    "fieldPath" : "metadata.name"
                  }
                }
              },
              {
                "name" : "RP_MODULES",
                "value" : "akka-cluster-bootstrapping,common,service-discovery"
              },
              {
                "name" : "RP_NAMESPACE",
                "value" : "akka-cluster-tooling-example"
              },
              {
                "name" : "RP_PLATFORM",
                "value" : "kubernetes"
              }
            ],
            "ports" : [
              {
                "containerPort" : 10000,
                "name" : "akka-remote"
              },
              {
                "containerPort" : 10001,
                "name" : "akka-mgmt-http"
              },
              {
                "containerPort" : 10002,
                "name" : "http"
              }
            ]
          }
        ]
      }
    }
  }
}
```

File: cli/src/main/scala/com/lightbend/rp/reactivecli/runtime/kubernetes/Deployment.scala [hunk 5]
```diff
@@ -273,62 +306,52 @@ object Deployment {
...
-                        "ports" -> annotations.endpoints.asJson)
-                        .deepmerge(annotations.readinessCheck.asJson(readinessProbeEncode))
-                        .deepmerge(annotations.healthCheck.asJson(livenessProbeEncode))).asJson))))))
-      case _ =>
-        Failure(new IllegalArgumentException("Unable to generate Kubernetes Deployment: both application name and version are required"))
+    deploymentType: DeploymentType): ValidationNel[String, Deployment] =
+
+    (annotations.appNameValidation |@| annotations.versionValidation) { (rawAppName, version) =>
+      val appName = serviceName(rawAppName)
+      val appNameVersion = serviceName(s"$appName$VersionSeparator$version")
```
Comment (fsat) [inline cli/src/main/scala/com/lightbend/rp/reactivecli/runtime/kubernetes/Deployment.scala]: This was the line that I missed!

**Pull request merged**